glean-core/ffi/src/lib.rs

```diff
@@ -35,8 +35,48 @@ use from_raw::*;
 use handlemap_ext::HandleMapExtension;
 use ping_type::PING_TYPES;
 
-lazy_static! {
-    static ref GLEAN: ConcurrentHandleMap<Glean> = ConcurrentHandleMap::new();
+pub(crate) fn with_glean<R, F>(callback: F) -> R::Value
+where
+    F: UnwindSafe + FnOnce(&Glean) -> Result<R, glean_core::Error>,
+    R: IntoFfi,
+{
+    let mut error = ffi_support::ExternError::success();
+    let res = ffi_support::abort_on_panic::call_with_result(&mut error, || {
+        let glean = glean_core::global_glean().lock().unwrap();
+        callback(&glean)
+    });
+    handlemap_ext::log_if_error(error);
+    res
+}
+
+pub(crate) fn with_glean_mut<R, F>(callback: F) -> R::Value
+where
+    F: UnwindSafe + FnOnce(&mut Glean) -> Result<R, glean_core::Error>,
+    R: IntoFfi,
+{
+    let mut error = ffi_support::ExternError::success();
+    let res = ffi_support::abort_on_panic::call_with_result(&mut error, || {
+        let mut glean = glean_core::global_glean().lock().unwrap();
+        callback(&mut glean)
+    });
+    handlemap_ext::log_if_error(error);
+    res
+}
+
+pub(crate) fn with_glean_value<R, F>(callback: F) -> R::Value
```

Comment:
Maybe add some docs about when to use `with_glean` vs. `with_glean_value`?

Reply:
Docs coming.